lib/mqttclient.ts

```diff
@@ -181,28 +181,138 @@ export default class MQTTClient {
+                                // Add this mapping to the instanceUUIDMapping array. This means
+                                // that all metrics that contain this instance path in their name
+                                // will have
+                                instanceUUIDMapping[instancePath] = metric.value;
+                            }
+                        }
+                    })
                 }
 
-                // If we've already seen this birth, update it
                 if (this.aliasResolver?.[topic.address.group]?.[topic.address.node]?.[topic.address.device]) {
-                    logger.info(`🔄 Received updated birth certificate for ${topic.address.group}/${topic.address.node}/${topic.address.device} with Instance_UUID ${instance}`);
+                    logger.info(`🔄 Received updated birth certificate for ${topic.address.group}/${topic.address.node}/${topic.address.device} with Instance_UUID ${topLevelInstance} using Schema_UUID ${topLevelSchema}`);
                 } else {
-                    logger.info(`👶 Received birth certificate for ${topic.address.group}/${topic.address.node}/${topic.address.device} with Instance_UUID ${instance}`);
+                    logger.info(`👶 Received birth certificate for ${topic.address.group}/${topic.address.node}/${topic.address.device} with Instance_UUID ${topLevelInstance} using Schema_UUID ${topLevelSchema}`);
                 }
 
                 // Store the birth certificate mapping in the alias resolver. This uses the alias as the key and a simplified object containing the name and type as the value.
                 this.setNestedValue(this.aliasResolver, [topic.address.group, topic.address.node, topic.address.device], payload.metrics.reduce(function (acc, obj) {
                     let alias = Long.isLong(obj.alias) ? obj.alias.toNumber() : obj.alias;
+
+                    // Work out all schemas that are involved in this metric.
+                    //
+                    // e.g. Assume the current metric is CNC/Axes/1/BaseAxis/Position/Actual
+                    //
+                    // The schemaUUIDMapping object will contain the following (non-relevant emitted):
+                    // - Axes/1: e39007e9-1427-4867-9d72-1c00c663db15
+                    // - Axes/1/BaseAxis: 777dd941-f426-4355-8130-e144530b1376
+                    // - Axes/1/BaseAxis/Position: 1a2c3594-d311-4f6b-865b-b97db3fa6d42
+
+                    // Schema_UUIDs
+
+                    let schemas = [];
+                    // Get all entries in the schemaUUIDMapping object that have keys that fit in the current obj.name
+                    Object.entries(schemaUUIDMapping).forEach(([schemaPath, schemaUUID]) => {
+                        // If the current obj.name contains the schemaPath then add the schemaUUID to the schemas array
+                        if (obj.name.includes(schemaPath)) {
+                            schemas.push(schemaUUID);
```

Comment:
This list of schemas comes out in random order; more specifically, it comes out in the order the Schema_UUID metrics happened to be listed in the BIRTH, rather than in path-down-the-tree order.

It's probably worth normalising the formatting of the UUIDs at this point, and verifying they are actually valid  UUIDs. Sparkplug transmits UUIDs as strings (stupid decision) so there is no guarantee about what those strings contain.